x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -163,6 +163,11 @@ public class ApiKeyService {
         TimeValue.timeValueHours(24L),
         Property.NodeScope
     );
+    public static final Setting<TimeValue> DELETE_RETENTION_PERIOD = Setting.positiveTimeSetting(
+        "xpack.security.authc.api_key.delete.retention_period",
+        TimeValue.timeValueDays(7),
+        Property.NodeScope
```

Comment:
Should this by dynamic? My gut says yes.

Reply:
Good point! Making it dynamic definitely helps with the concern of "breaking behaviour". I pushed https://github.com/elastic/elasticsearch/pull/92219/commits/bcd84d9560d13ae5457fcfcae7a03726bc7e7ca0 for it.

Reply:
It could be a better user story if we also make `delete.interval` dynamic. It should be a separate PR though.